bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValueImpl.java

```diff
@@ -229,9 +229,21 @@ public Object getRefValue(long index) {
         return get(index);
     }
 
+    @Override
+    public Object fillAndGetRefValue(long index) {
+        if (refValues != null) {
+            boolean needsFilling = index >= size;
+            if (needsFilling && this.elementType.getTag() == TypeTags.ARRAY_TAG) {
```

Comment:
```suggestion
            if (index >= size && this.elementType.getTag() == TypeTags.ARRAY_TAG) {
```

Reply:
Why are we specifically checking for array-typed elements though? This is valid for any array with an element type for which fill member is valid right?

The following seems to crash.

```ballerina
public function main() {
    map<int>[] arr = [];

    arr[0]["i"] = 1;
}
```

Reply:
I think we should also check if fill member is valid for this particular element type before calling `.getZeroValue()`.

The following also crashes
```ballerina
import ballerina/io;

public function main() {
    Foo[][2] arr = [];

    arr[0][1] = {s: ""}; // should fail here since Foo doesn't have a filler value

    io:println(arr[0][0].s); // fails since this particular `Foo` doesn't have a field `s`
}

type Foo record {
    string s;
};
```

ballerina-internal.log
```
[2020-03-10 11:46:37,334] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at abc.main(abc.bal:8)
	at ___init.$lambda$main$(.)
	at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:426)
	at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:218)
	at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:191)
	at java.lang.Thread.run(Thread.java:745)
 
[2020-03-10 11:46:37,339] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at abc.main(abc.bal:8)
	at ___init.$lambda$main$(.)
	at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:426)
	at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:218)
	at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:191)
	at java.lang.Thread.run(Thread.java:745)
```

Reply:
The `add()` method has the logic for checking if a particular type has a filler value. Will check and see what's going on here

Reply:
Fixed this by adding a separate fillRead() method instead of relying on the add() method. Some of the logic in the add() method wasn't really applicable to the fill-read case.